### DISCUSSION TITLE
no-unused-fragments while using codegen for typing

### DISCUSSION THREAD

**Discussion by lleger** [graphql-hive/graphql-eslint#1691] [Q&A]
Is it possible to use the [`no-unused-fragments`](https://the-guild.dev/graphql/eslint/rules/no-unused-fragments) eslint rule when using fragments for typing as espoused in [this article](https://the-guild.dev/blog/unleash-the-power-of-fragments-with-graphql-codegen) by the Guild?

We use codegen with fragments as described in the article but as our usage of fragments has grown we wanted to use the eslint rule to check that we're not leaving orphaned fragments that are no longer needed.

Unfortunately, the rule doesn't seem to detect fragments that are used in code and so it errors for every fragment used in this way.

I read the docs and did a bunch of searching and couldn't find anyone in a similar position, so wanted to ask!

**Comment by dimaMachina** [comment]
It works for definitions declared in code files, here are example https://github.com/B2o5T/graphql-eslint/tree/master/examples/code-file

**Reply by dimaMachina** [reply]
https://the-guild.dev/graphql/eslint/docs/getting-started#apply-plugin-to-graphql-definitions-defined-in-code-files

**Reply by lleger** [reply]
Admittedly I might be missing something but this didn't quite work for me. Here's what I'm seeing.

We have this fragment `getUserInitials.gql`:

```graphql
fragment GetUserInitialsUser on User {
  firstName
  lastName
}
```

We then use that fragment in a type:

```typescript
export const getUserInitials = (user: GetUserInitialsUserFragment) =>
  `${user.firstName.charAt(0)}${user.lastName.charAt(0)}`.toUpperCase();
```

The `GetUserInitialsUserFragment` is generated by codegen.

eslint is throwing a warning with `no-unused-fragments` because it's not used in graphql but it is being used.

**Reply by dimaMachina** [reply]
`GetUserInitialsUser` should be used in graphql sdl

more detailed info on why written here https://stackoverflow.com/a/62890539

**Reply by lleger** [reply]
Is the takeaway just that using fragments outside of the actual requests just not supported? We were essentially using the fragments to generates types like `Pick<User, "__typename" | "firstName" | "lastName">` since for very complex types writing the type by hand can be cumbersome (a bit of a simple example here, but with nesting it can get quite complex to write). But that is perhaps us abusing the codegen tool in a way it's not intended?

**Reply by dimaMachina** [reply]
please share reproduction repo and I'll take a look if I could enhance rule for this case
